addons/test_mail_full/tests/test_mail_thread_phone.py

```diff
@@ -65,3 +65,18 @@ def test_search_phone_mobile_search(self):
         results = self.env['mail.test.sms.bl'].search([('phone_mobile_search', 'not ilike', '0475')])
         results = self.env['mail.test.sms.bl'].search([('phone_mobile_search', 'not ilike', '101')])
         results = self.env['mail.test.sms.bl'].search([('phone_mobile_search', 'not ilike', '606')])
+
+    @users('employee')
+    def test_search_phone_mobile_search_boolean(self):
```

Comment:
the test doesn't check for the "and" or the "or" operator 
this line:
```
op = '&' if final_operator == '=' else '|'
```

(could be nice to set `mobile_nbr` to False on one of the records)